src/ralph/models/selector.py

```diff
@@ -64,20 +65,36 @@ class ModelSelector:
         decision_tree (dict): Stores the rule checking order for model selection.
     """
 
-    model_rules: dict[Type[BaseModel], list[Rule]] = {}
-
     def __init__(self, module="ralph.models.edx"):
         """Instantiates ModelSelector."""
 
+        self.model_rules: dict[Type[BaseModel], list[Rule]] = {}
         for object_ in import_module(module).__dict__.values():
             if (
                 isinstance(object_, type)
                 and issubclass(object_, BaseModel)
                 and hasattr(object_, "__selector__")
             ):
-                self.model_rules[object_] = object_.__selector__
+                self.add_to_model_rules(object_, object_.__selector__)
         self.decision_tree = self.get_decision_tree(self.model_rules)
 
+    def add_to_model_rules(self, new_model: Type[BaseModel], new_rules: list[Rule]):
+        """Adds a new model with it's list of rules to the model_rules dictionary.
+
+        Raises:
+            ModelRulesException: When new_rules list is a subset or superset of another rules list.
+        """
+
+        rules_set = set(new_rules)
```

Comment:
Let's consider that rules are always sets:

```suggestion
        new_rules = set(new_rules)
```

Reply:
Good point) I would really like to use sets for the rules (tried twice) but unfortunately the output of the decision tree depends on the order of the rules. see https://github.com/openfun/ralph/pull/66#discussion_r594695806
The output with rules in a different order will be equally right (Edge case: when two (or more) rules have the same number of occurrences) but different - which breaks the tests :disappointed:

Reply:
I just wanted to use sets for rules in this method, not generally.